### PR TITLE
LPAL-629 remove references to Sendgrid to ease the path to PHP 8

### PR DESCRIPTION
**Pull request by nickdavis2001**
## Purpose

_Now that we're on Notify and its tried and tested, remove references to Sendgrid to ease the path to PHP 8. This only affects service-front_

## Approach

_Mostly deletions_ 

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/module/Application/src/Model/Service/Mail/Transport/MailTransportFactory.php [hunk 1]
```diff
@@ -40,18 +39,7 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
...
-            if (!isset($sendGridConfig['key'])) {
-                throw new RuntimeException('Sendgrid settings not found');
-            }
-
-            $client = new SendGrid($sendGridConfig['key']);
-            $messageFactory = $container->get('MessageFactory');
-
-            return new SendGridMailTransport($client->client, $messageFactory);
-        } elseif ($transport === 'notify') {
+        if ($transport === 'notify') {
```
Comment (Raffers) [inline service-front/module/Application/src/Model/Service/Mail/Transport/MailTransportFactory.php:42]: Wouldn't we want to remove this completely? Along with removing the very end RuntimeException. As we shouldn't be dependent on config to make sure notify is specified.

**Review by Raffers — CHANGES_REQUESTED** [review]

File: service-front/module/Application/src/Model/Service/Mail/Transport/MailTransportFactory.php (hunk 1 above)
Comment (nickdavis2001) [reply service-front/module/Application/src/Model/Service/Mail/Transport/MailTransportFactory.php:42]: Agreed. Somehow that slipped through the net, will take that out too

File: service-front/config/autoload/global.php [hunk 2]
```diff
@@ -123,14 +123,7 @@
...
         // implementation-specific configuration
-        'transport' => getenv('OPG_LPA_FRONT_EMAIL_TRANSPORT') ?: 'sendgrid',
-
-        'sendgrid' => [
-            'key'     => getenv('OPG_LPA_FRONT_EMAIL_SENDGRID_API_KEY') ?: null,
-            'webhook' => [
-                'token' => getenv('OPG_LPA_FRONT_EMAIL_SENDGRID_WEBHOOK_TOKEN') ?: null,
-            ],
-        ], //sendgrid
+        'transport' => getenv('OPG_LPA_FRONT_EMAIL_TRANSPORT') ?: null,
```
Comment (Raffers) [inline service-front/config/autoload/global.php:126]: Can also remove this & all reference to `OPG_LPA_FRONT_EMAIL_TRANSPORT` now we've removed that logic

Comment (nickdavis2001) [reply service-front/config/autoload/global.php]: Done

**Review by Raffers — DISMISSED** [review]

**Review by Raffers — APPROVED** [review]

**Review by williamfalconeruk — APPROVED** [review]

**Pull request merged**